src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1279,6 +1279,7 @@ void Visitor::setMatchingWordAndScoreVisibleIfPresent(
     }
     for (std::string_view s : std::vector<std::string>(
              absl::StrSplit(name.substr(1, name.size() - 2), ' '))) {
+      addVisibleVariable(var->getScoreVariable(std::string(s)));
```

Comment:
Don't forget to drop the `std::string(...)` again once you have a dedicated function for the `getWordScoreVariable`.
And rename the `getScoreVariable` to `getEntityScoreVariable` or something.